README.md

```diff
@@ -277,6 +277,147 @@ MotionAnimator is a purely Core Animation-based animator. If you are looking for
+
+A subset of UIView's and CALayer's public APIs is animatable by Core Animation. Of these animatable properties, some are implicitly animatable while some are not. Whether a property is animatable or not depends on the context within which it's being animated, and whether an animation is additive or not depends on which animation API is being used. With this matrix of conditions it's understandable that it can sometimes be difficult to know how to effectively make use of Core Animation.
+
+The following quiz helps illustrate that the UIKit and Core Animation APIs can often lead to unintuitive behavior. Try to guess which of the following snippets will generate an animation and, if they do, what the generated animation's duration will be:
+
+> Imagine that each code snippet is a standalone unit test (because [they are](tests/unit/HeadlessLayerImplicitAnimationTests.swift)!).
+
+```swift
+let view = UIView()
+UIView.animate(withDuration: 0.8, animations: {
+  view.alpha = 0.5
+})
+```
+
+<details>
+  <summary>Click to see the answer</summary>
+  Generates an animation with duration of 0.8.
+</details>
+
+---
+
+```swift
+let view = UIView()
+UIView.animate(withDuration: 0.8, animations: {
+  view.layer.opacity = 0.5
+})
+```
+
+<details>
+  <summary>Click to see the answer</summary>
+  Generates an animation with duration of 0.8.
+</details>
+
+---
+
+```swift
+let view = UIView()
+UIView.animate(withDuration: 0.8, animations: {
+  view.layer.cornerRadius = 3
+})
+```
+
+<details>
+  <summary>Click to see the answer</summary>
+  On iOS 11 and up, generates an animation with duration of 0.8. Older operating systems will not generate an animation.
+</details>
+
+---
+
+```swift
+let view = UIView()
+view.alpha = 0.5
+```
+
+<details>
+  <summary>Click to see the answer</summary>
+  Does not generate an animation.
+</details>
+
+---
+
+```swift
+let view = UIView()
+view.layer.opacity = 0.5
+```
+
+<details>
+  <summary>Click to see the answer</summary>
+  Does not generate an animation.
+</details>
+
+---
+
+```swift
+let layer = CALayer()
+layer.opacity = 0.5
+```
+
+<details>
+  <summary>Click to see the answer</summary>
+  Does not generate an animation.
+</details>
+
+---
+
+```swift
+let view = UIView()
+window.addSubview(view)
+let layer = CALayer()
+view.layer.addSublayer(layer)
+
+// Pump the run loop once.
+RunLoop.main.run(mode: .defaultRunLoopMode, before: .distantFuture)
+
+layer.opacity = 0.5
+```
+
+<details>
+  <summary>Click to see the answer</summary>
+  Generates an animation with duration of 0.25.
+</details>
+
+---
+
+```swift
+let view = UIView()
+window.addSubview(view)
+let layer = CALayer()
+view.layer.addSublayer(layer)
+
+// Pump the run loop once.
+RunLoop.main.run(mode: .defaultRunLoopMode, before: .distantFuture)
+
+UIView.animate(withDuration: 0.8, animations: {
+  layer.opacity = 0.5
+})
+```
+
+<details>
+  <summary>Click to see the answer</summary>
+  Generates an animation with duration of 0.25.
```

Comment:
Is 0.25 correct here?

Reply:
Yep!

```swift

func testDoesImplicitlyAnimateInUIViewAnimateBlock() {
  UIView.animate(withDuration: 0.5) {
    self.layer.opacity = 0.5
  }

  let animation = layer.animation(forKey: "opacity") as! CABasicAnimation
  XCTAssertEqual(animation.keyPath, "opacity")
  XCTAssertEqual(animation.duration, CATransaction.animationDuration())
}
```

Super fun, right?

Reply:
Maybe add a not that makes it explicit that the docs are correct? I suspect most people would just think we have an error in the README here.

Reply:
Ah, good call. Will clarify.